backend/hct_mis_api/apps/account/admin/partner.py

```diff
@@ -12,21 +12,31 @@
 from admin_extra_buttons.decorators import button
 
 from hct_mis_api.apps.account import models as account_models
-from hct_mis_api.apps.account.models import IncompatibleRoles, PartnerPermission, Role
-from hct_mis_api.apps.core.models import BusinessArea
+from hct_mis_api.apps.account.models import IncompatibleRoles, Role
+from hct_mis_api.apps.core.models import BusinessArea, BusinessAreaPartnerThrough
 from hct_mis_api.apps.geo.models import Area
 from hct_mis_api.apps.program.models import Program
 from hct_mis_api.apps.utils.admin import HopeModelAdminMixin
 from mptt.forms import TreeNodeMultipleChoiceField
 
+if TYPE_CHECKING:
+    from django.db.models.query import QuerySet
+
 
 def can_add_business_area_to_partner(request: Any, *args: Any, **kwargs: Any) -> bool:
     return request.user.can_add_business_area_to_partner()
 
 
-class BusinessAreaRoleForm(forms.Form):
-    business_area = forms.ModelChoiceField(queryset=BusinessArea.objects.all(), required=True)
-    roles = forms.ModelMultipleChoiceField(queryset=Role.objects.all(), required=True)
+def business_area_role_form_custom_query(queryset: "QuerySet") -> Any:
+    class BusinessAreaRoleForm(forms.Form):
+        business_area = forms.ModelChoiceField(queryset=BusinessArea.objects.all(), required=True)
+        roles = forms.ModelMultipleChoiceField(queryset=Role.objects.all(), required=True)
+
+        def __init__(self, *args: Any, **kwargs: Any) -> None:
+            super().__init__(*args, **kwargs)
+            self.fields["business_area"].queryset = queryset
```

Comment:
Not sure if I understand this construction - queryset for business_area is declared above, but it can be overwritten by param from function? I would see rather queryset as param in __init__, then you can do `MyForm(queryset=...)`

Reply:
The issue is that we are passing a class to `formset_factory`, not the object.

Reply:
So I did it in a bit of a hacky way, as I need to filter BAs based on the param